### PR TITLE
Adding Cognito user creation jobs

### PR DESCRIPTION
**Pull request by ChrisScottThomas**
Created Concourse jobs to create users in Cognito, using AWS SecretsManager to store user content.

### CODE REVIEW

File: ci/jobs/mgmt-dev-user-creation.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+jobs:
+- name: management-dev-user-creation
+  plan:
+  - get: aws-concourse-users
+    trigger: true
+  - .: (( inject meta.plan.create-aws-profiles ))
+    config:
+      params:
+        AWS_ACC: "{{accounts['management-dev']}}"
```
Comment (dan-hill2802) [inline ci/jobs/mgmt-dev-user-creation.yml.j2:9]: why not grab straight from secret store?

File: ci/resources.yml [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
-    access_token: ((github_com.token))
-  check_every: 720h
-
+  - name: aws-concourse-pr
+    type: pull-request
+    source:
+      access_token: ((github_com.token))
+      repo: dwp/aws-concourse
+      uri: https://github.com/dwp/aws-concourse.git
+    check_every: 15m
```
Comment (dan-hill2802) [inline ci/resources.yml:8]: should be 720 hours + a webhook

File: ci/resources.yml [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
+  - name: aws-concourse
+    type: git
+    source:
+      branch: master
+      username: ((github_com.username))
+      password: ((github_com.token))
+      uri: https://github.com/dwp/aws-concourse.git
+    check_every: 15m
```
Comment (dan-hill2802) [inline ci/resources.yml:16]: should be 720 hours + a webhook

File: ci/resources.yml [hunk 4]
```diff
@@ -1,24 +1,33 @@
...
+    check_every: 720h
+  - name: aws-concourse-users
+    type: git
+    source:
+      branch: master
+      private_key: ((svc-dip.private_key))
+      uri: git@github.ucds.io:dip/dataworks-secrets.git
+      path:
+        - concourse/dataworks/concourse-cognito/**
+    check_every: 720h
```
Comment (dan-hill2802) [inline ci/resources.yml:32]: needs to be something smaller like 5m

File: ci/resources.yml (hunk 2 above)
Comment (dan-hill2802) [reply ci/resources.yml:8]: wrong concourse

File: ci/resources.yml (hunk 3 above)
Comment (dan-hill2802) [reply ci/resources.yml:16]: wrong concourse

File: ci/resources.yml (hunk 4 above)
Comment (dan-hill2802) [reply ci/resources.yml:32]: wrong concourse

File: ci/meta.yml [hunk 5]
```diff
@@ -320,3 +320,80 @@ meta:
...
+        image_resource:
+          type: docker-image
+          source:
+            repository: ((docker-awscli.repository))
+            version: ((docker-awscli.version))
+            tag: ((docker-awscli.version))
+        params:
+          AWS_PROFILE: ci
+          AWS_SHARED_CREDENTIALS_FILE: .aws/credentials
+          AWS_CONFIG_FILE: .aws/config
```
Comment (mattburgess) [inline ci/meta.yml:337]: Why are we going back to this "old" pattern, and not following the `source /assume-role` pattern as per https://git.ucd.gpn.gov.uk/dip/aws-common-infrastructure/wiki/AWS-Concourse-Pipeline-Migration-Guide#on-the-subject-of-credentials? (admittedly, that doc could do with a fully-fleshed example!).

Comment (dan-hill2802) [reply ci/meta.yml:337]: this is running on Crown Concourse

Comment (ChrisScottThomas) [reply ci/meta.yml:337]: > this is running on Crown Concourse

:this:

**Pull request merged**